packages/dara-core/js/api/websocket.tsx

```diff
@@ -258,48 +182,106 @@ export class WebSocketClient implements WebSocketClientInterface {
 
     closeHandler: () => void;
 
-    constructor(_socket: WebSocket, _token: string, _liveReload = false) {
-        this.socket = _socket;
+    _maxAttempts: number;
+
+    _pingInterval: NodeJS.Timeout;
+
+    _socketUrl: string;
+
+    _reconnectCount: number;
+
+    constructor(_socketUrl: string, _token: string, _liveReload = false) {
         this.token = _token;
         this.liveReload = _liveReload;
         this.messages$ = new Subject();
         this.closeHandler = this.onClose.bind(this);
-        this.initialize();
+        this._socketUrl = _socketUrl;
+        this._reconnectCount = 0;
+        this._pingInterval = null;
+        this._maxAttempts = maxAttempts;
+
+        // Lastly call initialize to setup the socket properly
+        this.socket = this.initialize();
     }
 
-    initialize(): void {
+    initialize(): WebSocket {
```

Comment:
I might be missing something but I think this implementation doesn't implement liveReload at all?
Could we add a test case to ensure `location.reload` is invoked on reconnect if `liveReload` is true?

Reply:
Ah, good catch. Sorry I took that out for simplicity when testing everything but forgot to put it back. I'll readd and add a test.

Reply:
Should be sorted now, let me know if you can think of any issues.